src/test/java/eu/elixir/ega/ebi/keyproviderservice/config/MyCipherConfigTest.java

```diff
@@ -0,0 +1,267 @@
+
+/**
+ * Test class for {@link MyCipherConfig}.
+ * 
+ * @author amohan
+ */
+@RunWith(SpringRunner.class)
+@TestPropertySource(locations = "classpath:application-test.properties")
+public class MyCipherConfigTest {
+
+    public final static Long KEY_ID = new Long("-6324474004599780284");
+    public static ClassLoader classLoader;
+    public static File file;
+    public static File filePassPath;
+    public static File fileOutput;
+
+    @Autowired
+    MyCipherConfig myCipherConfig;
+
+    @MockBean
+    private RestTemplate restTemplate;
+
+    /**
+     * Test {@link MyCipherConfig#getPrivateKey(Long)}. Verify the PGPPrivateKey
+     * keyid.
+     */
+    @Test
+    public void testGetPrivateKey() {
+        PGPPrivateKey pgpPrivateKeys = myCipherConfig.getPrivateKey(KEY_ID);
+        assertThat(pgpPrivateKeys.getKeyID(), equalTo(KEY_ID));
+    }
+
+    /**
+     * Test {@link MyCipherConfig#getPublicKey(Long)}. Verify the PGPPrivateKey
+     * keyid.
+     */
+    @Test
+    public void testGetPublicKey() {
+        PGPPublicKey pgpPublicKey = myCipherConfig.getPublicKey(KEY_ID);
+        assertThat(pgpPublicKey.getKeyID(), equalTo(KEY_ID));
+    }
+
+    /**
+     * Test {@link MyCipherConfig#getKeyPaths(Long)}. Verify the keyPath and
+     * keyPassPath.
+     */
+    @Test
+    public void testGetKeyPaths() {
+        KeyPath keyPath = myCipherConfig.getKeyPaths(KEY_ID);
+        assertThat(keyPath.getKeyPath(), equalTo(file.getAbsolutePath()));
+        assertThat(keyPath.getKeyPassPath(), equalTo(filePassPath.getAbsolutePath()));
+
+    }
+
+    /**
+     * Test {@link MyCipherConfig#getKeyPaths(Long)}. Verify the asciiArmouredKey
+     * not null. We cannot check the content of asciiArmouredKey as it is
+     * reArmoured.
+     */
+    @Test
+    public void testGetAsciiArmouredKeys() {
+        String asciiArmouredKey = myCipherConfig.getAsciiArmouredKey(KEY_ID);
+        assertNotNull(asciiArmouredKey);
+    }
+
+    /**
+     * Test {@link MyCipherConfig#getKeyPaths(Long)}. Verify size of the keyIds.
+     */
+    @Test
+    public void testGetKeyIDs() {
+        Set<Long> keyIDs = myCipherConfig.getKeyIDs();
+        assertThat(keyIDs.iterator().next(), equalTo(KEY_ID));
+        assertThat(keyIDs.size(), equalTo(1));
+    }
+
+    /**
+     * Test {@link MyCipherConfig#readFileAsString(String)}. Verify size of the
+     * privateKey read from file is same as expected.
+     */
+    @Test
+    public void testReadFileAsString() throws IOException {
+        String privateKey = myCipherConfig.readFileAsString(file.getAbsolutePath());
+        assertThat(privateKey, equalTo(getPrivateKey()));
+    }
+
+    /**
+     * Test {@link MyCipherConfig#readFileAsString(String)}. Verify the
+     * pgpPrivateKey.
+     */
+    @Test
+    public void testExtractKey() throws IOException {
+        PGPPrivateKey pgpPrivateKey = myCipherConfig.extractKey(getPrivateKey(), getPassphrase());
+        assertNotNull(pgpPrivateKey);

```

Comment:
Is it possible to verify that extracted key is correct instead just checking that it isn't NULL?